src/commands/generate_buildpack_matrix/command.rs

```diff
@@ -1,34 +1,70 @@
-use crate::buildpacks::{find_releasable_buildpacks, read_image_repository_metadata};
+use crate::buildpacks::{
+    find_releasable_buildpacks, read_buildpack_descriptor, read_image_repository_metadata,
+};
 use crate::commands::generate_buildpack_matrix::errors::Error;
+use crate::commands::resolve_path;
 use crate::github::actions;
 use clap::Parser;
-use libcnb_package::{
-    get_buildpack_target_dir, read_buildpack_data, BuildpackData, GenericMetadata,
+use libcnb_data::buildpack::{BuildpackDescriptor, BuildpackId};
+use libcnb_package::output::{
+    create_packaged_buildpack_dir_resolver, default_buildpack_directory_name,
 };
+use libcnb_package::CargoProfile;
 use std::collections::{BTreeMap, HashSet};
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Parser, Debug)]
 #[command(author, version, about = "Generates a JSON list of buildpack information for each buildpack detected", long_about = None)]
-pub(crate) struct GenerateBuildpackMatrixArgs;
+pub(crate) struct GenerateBuildpackMatrixArgs {
+    #[arg(long)]
+    pub(crate) working_dir: Option<PathBuf>,
+    #[arg(long)]
+    pub(crate) package_dir: PathBuf,
+    #[arg(long)]
+    pub(crate) release: Option<bool>,
+    #[arg(long, default_value = "x86_64-unknown-linux-musl")]
+    pub(crate) target: String,
+}
+
+pub(crate) fn execute(args: GenerateBuildpackMatrixArgs) -> Result<()> {
+    let working_dir = std::env::current_dir()
+        .map(|base| match args.working_dir {
+            Some(path) => resolve_path(&path, &base),
+            None => base,
```

Comment:
The match here could be replaced by `args.working_dir.map_or(...)`

(and similarly for the other usages of this pattern in the other `execute()` methods)